readme.md

```diff
@@ -1,100 +1,147 @@
-What is it
-===========
+==========
+What it is
+==========
 
-It is a prerendering server based on PhantomJS. It loads JS based app remotely and sends already processed page with JS cut out.
+It is a prerendering server based on PhantomJS. It loads JS based app
+remotely and sends already processed page with JS cut out.
 
-The idea is based on [angular-seo](https://github.com/steeve/angular-seo)
-But the actual code was changed to fit our GoAbout app. However, the renderer is app-agnostic adn it can be used with any JS-based app, not only ours. You just need to install it on your own machine.
+The idea is based on [angular-seo](https://github.com/steeve/angular-
+seo) But the actual code was changed to fit our GoAbout app. However,
+the renderer is app-agnostic and it can be used with any JS-based app,
+not only ours. You just need to install it on your own machine.
 
 
 How to use with AngularJS (or any other framework)
-=======
-The solution is made of 3 parts:
-- small modification of your static HTML file
-- an AngularJS module, that you have to include and call
-- PhantomJS script
+==================================================
+
+The solution is made of 4 parts:
+
+1. small modification of your static HTML file [optional]
+2. an AngularJS module, that you have to include and call
+3. PhantomJS script
+4. redirect request from bots to PhantomJS
+
+1. Modifying your static HTML
+-----------------------------
 
-Modifying your static HTML
-==========================
+The `#!` escape for browsers that use HTML5 push-state makes crawlers
+request your page as
+`http://your.site/page?_escape_fragment_=some/push/state/url. If you
```

Comment:
Ok. I'll include a link to escape fragments.

The thing is, that this is only needed if your app uses push-state without `#!`,.. right?